bluez-async/src/lib.rs

```diff
@@ -696,6 +712,12 @@ impl BluetoothSession {
         self.await_service_discovery(id).await
     }
 
+    /// Connect to the given Bluetooth device with specified timeout.
+    pub async fn connect_with_timeout(&self, id: &DeviceId, timeout: Duration) -> Result<(), BluetoothError> {
+        self.device_with_timeout(id, timeout).connect().await?;
```

Comment:
Does this actually work as expected when you test it with real devices? You're changing the D-Bus method call timeout for the `connect` method, but I thought BlueZ still had its own timeout separate from that.

Reply:
Yes, I checked in debug that 
```
fn device_with_timeout(&self, id: &DeviceId, timeout: Duration)...........
        Proxy::new(
            "org.bluez",
            id.object_path.to_owned(),
            local_timeout_value,
```
is really called with custom value from btleplug - > discover_adapters_peripherals ->
```
                if !is_connected {
                    println!("Connecting to peripheral {:?}...", &local_name);
                    if let Err(err) = peripheral.connect_with_timeout(Duration::from_secs(20)).await {

```
it works as expected.